src/Etterna/Actor/Menus/MusicWheel.cpp

```diff
@@ -1763,6 +1777,28 @@ class LunaMusicWheel : public Luna<MusicWheel>
 		LuaHelpers::ReadArrayFromTable(newHashList, L);
 		lua_pop(L, 1);
 		p->SetHashList(newHashList);
+		
+		std::vector<string> newOutHashList;
+		p->SetOutHashList(newOutHashList);
+
+		p->ReloadSongList(false, "");
+	}
+
+	static auto FilterByAndAgainstStepKeys(T* p, lua_State* L) -> int
+	{
+		luaL_checktype(L, 1, LUA_TTABLE);
+		lua_pushvalue(L, 1);
+		std::vector<string> newHashList;
+		LuaHelpers::ReadArrayFromTable(newHashList, L);
+		lua_pop(L, 1);
+		p->SetHashList(newHashList);
+		luaL_checktype(L, 2, LUA_TTABLE);
+		lua_pushvalue(L, 2);
+		std::vector<string> newOutHashList;
+		LuaHelpers::ReadArrayFromTable(newOutHashList, L);
+		lua_pop(L, 1);
+		p->SetOutHashList(newOutHashList);
+
 		p->ReloadSongList(false, "");
 		return 1;
 	}

```

Comment:
I didn't quite understand the lua documentation on how lua call stack works, so please make sure to review these lines.

Reply:
This looks fine to me, untested, but I feel that unexpected behavior may arise when sending a non-table second parameter to this function. SetHashList may execute and then the function would error out on the next line.

Reply:
This is only ever meant to be called with 2 tables as parameters, never more, never less. That's why I kept FilterByStepKeys (mostly) intact in case anyone wanted that functionality without plugging in an additional value.

Reply:
I'm inclined to guess that the parameter checks are runtime especially in those helper functions. So if you pass a nil for whatever reason, like as an accident, then the function would not exit until it reaches the `luaL_checktype(L, 2, LUA_TTABLE)` where it exits and reports the error to the game.
So what I'm saying is the parameters should be taken in and placed in those variables as the first thing rather than immediately be used and thrown away.

Reply:
However upon testing this with breakpoints real quick it seems to know that ahead of time so I'll call it good.
Didn't want to make the assumption though.

Reply:
so something like
```
static auto FilterByAndAgainstStepKeys(T* p, lua_State* L) -> int
{
	luaL_checktype(L, 1, LUA_TTABLE);
	luaL_checktype(L, 2, LUA_TTABLE);
	lua_pushvalue(L, 1);
	lua_pushvalue(L, 2);
	std::vector<string> newHashList;
	LuaHelpers::ReadArrayFromTable(newHashList, L);
	lua_pop(L, 1);
	std::vector<string> newOutHashList;
	LuaHelpers::ReadArrayFromTable(newOutHashList, L);
	lua_pop(L, 1);

	p->SetHashList(newHashList);
	p->SetOutHashList(newOutHashList);

	p->ReloadSongList(false, "");
	return 1;
}
```
would be better in your opinion then?

Reply:
```cpp
	static auto FilterByAndAgainstStepKeys(T* p, lua_State* L) -> int
	{
		luaL_checktype(L, 1, LUA_TTABLE);
		lua_pushvalue(L, 1);
		std::vector<string> newHashList;
		LuaHelpers::ReadArrayFromTable(newHashList, L);
		lua_pop(L, 1);
		luaL_checktype(L, 2, LUA_TTABLE);
		lua_pushvalue(L, 2);
		std::vector<string> newOutHashList;
		LuaHelpers::ReadArrayFromTable(newOutHashList, L);
		lua_pop(L, 1);

		p->SetHashList(newHashList);
		p->SetOutHashList(newOutHashList);
		p->ReloadSongList(false, "");

		return 1;
	}
```

Reply:
OK I see so that the function doesn't call anything external before the parameters are set up.
Sure I'll change that then.